contracts/DCAHubSwapper/ThirdPartyDCAHubSwapper.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.8.7 <0.9.0;
+
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '@mean-finance/dca-v2-core/contracts/interfaces/IDCAHubSwapCallee.sol';
+
+contract ThirdPartyDCAHubSwapper is IDCAHubSwapCallee {
+  /// @notice A target we want to give allowance to
+  struct Allowance {
+    IERC20 token;
+    address spender;
+    uint256 amount;
+  }
+
+  /// @notice The data necessary for a swap to be executed
+  struct SwapExecution {
+    address swapper;
+    bytes swapData;
+  }
+
+  /// @notice Data used for the callback
+  struct SwapWithDexesCallbackData {
+    // Timestamp where the tx is no longer valid
+    uint256 deadline;
+    // Targets to set allowance to
+    Allowance[] allowanceTargets;
+    // The different swaps to execute
+    SwapExecution[] executions;
+    // A list of tokens to check for unspent balance (should not be reward/to provide)
+    IERC20[] intermediateTokensToCheck;
+    // The address that will receive the unspent tokens
+    address leftoverRecipient;
+    // This flag is just a way to make transactions cheaper. If Mean Finance is executing the swap, then it's the same for us
+    // if the leftover tokens go to the hub, or to another address. But, it's cheaper in terms of gas to send them to the hub
+    bool sendToProvideLeftoverToHub;
+  }
+
+  /// @notice Thrown when deadline has passed
+  error TransactionTooOld();
+
+  using SafeERC20 for IERC20;
+  using Address for address;
+
+  // solhint-disable-next-line func-name-mixedcase
+  function DCAHubSwapCall(
+    address,
+    IDCAHub.TokenInSwap[] calldata _tokens,
+    uint256[] calldata,
+    bytes calldata _data
+  ) external {
+    SwapWithDexesCallbackData memory _callbackData = abi.decode(_data, (SwapWithDexesCallbackData));
+    if (block.timestamp > _callbackData.deadline) revert TransactionTooOld();
```

Comment:
Do you think we really need this? Most of the trades are going to be routed through places that already have a deadline check.

Reply:
Only a few due, and by checking it ourselves we can fail fast (and cheap)